meshmode/dof_array.py

```diff
@@ -277,7 +277,21 @@ def __getstate__(self):
         if self.array_context is not actx:
             ary = _thaw(actx, _freeze(self))
 
-        return [actx.to_numpy(ary_i) for ary_i in ary._data]
+        d = {}
+        d["data"] = [actx.to_numpy(ary_i) for ary_i in ary._data]
+
+        if len(ary) > 0 and hasattr(ary._data[0], "tags"):
+            d["tags"] = [ary_i.tags for ary_i in ary._data]
+        else:
+            d["tags"] = [frozenset() for _ in range(len(ary._data))]
```

Comment:
```suggestion
	d["tags"] = [getattr(ary_i, "tags", frozenset()) for ary_i in ary]
```
This looks more readable to me and it should catch all the cases. What do you think?

Reply:
There's probably a similar rewrite for the `axes`, but that's not going to look as nice because of the nesting :\

Reply:
Looks good, thanks!